src/RingGrids/general.jl

```diff
@@ -0,0 +1,324 @@
+    colats, lons = get_colatlons(Grid, nlat_half)   # colatitudes, longitudes in radians
+    lats = colats                                   # flat copy rename before conversion
+    lats .= π/2 .- colats                           # colatiudes to latitudes in radians
+    return lats, lons
+end
+
+get_lat(grid::Grid) where {Grid<:AbstractGridArray} = get_lat(Grid, grid.nlat_half)
+get_latd(grid::Grid) where {Grid<:AbstractGridArray} = get_latd(Grid, grid.nlat_half)
+get_lond(grid::Grid) where {Grid<:AbstractGridArray} = get_lond(Grid, grid.nlat_half)
+get_lon(grid::Grid) where {Grid<:AbstractGridArray} = get_lon(Grid, grid.nlat_half)
+get_colat(grid::Grid) where {Grid<:AbstractGridArray} = get_colat(Grid, grid.nlat_half)
+get_colatlons(grid::Grid) where {Grid<:AbstractGridArray} = get_colatlons(Grid, grid.nlat_half)
+get_nlon_max(grid::Grid) where {Grid<:AbstractGridArray} = get_nlon_max(Grid, grid.nlat_half)
+
+function get_lat(Grid::Type{<:AbstractGridArray}, nlat_half::Integer)
+    return π/2 .- get_colat(Grid, nlat_half)
+end
+
+function get_latd(Grid::Type{<:AbstractGridArray}, nlat_half::Integer)
+    return get_lat(Grid, nlat_half) * (360/2π)
+end
+
+# only defined for full grids, empty vector as fallback
+get_lon(::Type{<:AbstractGridArray}, nlat_half::Integer) = Float64[]
+get_lond(::Type{<:AbstractGridArray}, nlat_half::Integer) = Float64[]
+
+"""
+$(TYPEDSIGNATURES)
+Returns a vector `nlons` for the number of longitude points per latitude ring, north to south.
+Provide grid `Grid` and its resolution parameter `nlat_half`. For both_hemisphere==false only
+the northern hemisphere (incl Equator) is returned."""
+function get_nlons(Grid::Type{<:AbstractGridArray}, nlat_half::Integer; both_hemispheres::Bool=false)
+    n = both_hemispheres ? get_nlat(Grid, nlat_half) : nlat_half
+    return [get_nlon_per_ring(Grid, nlat_half, j) for j in 1:n]
+end
+
+## ITERATORS
+"""
+$(TYPEDSIGNATURES)
+CartesianIndices for the 2nd to last dimension of an AbstractGridArray,
+to be used like
+
+for k in eachgrid(grid)
+    for ring in eachring(grid)
+        for ij in ring
+            grid[ij, k]"""
+@inline eachgrid(grid::AbstractGridArray) = CartesianIndices(size(grid)[2:end])
+
+"""
+$(TYPEDSIGNATURES)
+Vector{UnitRange} `rings` to loop over every ring of grid `grid`
+and then each grid point per ring. To be used like
+
+    rings = eachring(grid)
+    for ring in rings
+        for ij in ring
+            grid[ij]"""
+@inline eachring(grid::AbstractGridArray) = grid.rings
+
+function eachring(Grid::Type{<:AbstractGridArray}, nlat_half::Integer)
+    rings = Vector{UnitRange{Int}}(undef, get_nlat(Grid, nlat_half))
+    each_index_in_ring!(rings, Grid, nlat_half)
+    return rings
+end
+
+"""
+$(TYPEDSIGNATURES)
+Same as `eachring(grid)` but performs a bounds check to assess that all grids
+in `grids` are of same size."""
+function eachring(grid1::Grid, grids::Grid...) where {Grid<:AbstractGridArray}
+    @inline 
```

Comment:
The `@inline` in an individual line is fine?

Reply:
Yes, I remember I put it there deliberately for some performance reasons, but it doesn't seem to be needed anymore. Technically it's a callsite `@inline` which is also possible since some Julia version. I removed it now, the code runs at 2ns with or without